### PR TITLE
Support multiple VPCs for a route53_zone

### LINKED ISSUE #1299: Support multiple VPCs for a route53_zone
### Summary

Similar to this old issue (https://github.com/ansible/ansible/issues/19114) on the old repo, I'd like to be able to create a route53 zone with multiple VPCs. I understand the boto functionality is there now such that you can achieve this (albeit with multiple calls). This should be more possible now

### Issue Type

Feature Idea

### Component Name

route53_zone

### Additional Information
I want to run something like

```yaml (paste below)
- name: create a private zone across 2 VPCs
  amazon.aws.route53_zone:
    zone: devel.example.com
    vpcs:
      - id: vpc-123456
        region: us-west-2
      - id: vpc-000001
        region: us-west-2
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Tompage1994**
##### SUMMARY

Added support for multiple VPCs to be applied to route53_zone module
Fixes #1299 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

route53_zone

##### ADDITIONAL INFORMATION

```paste below
    - name: create zone
      amazon.aws.route53_zone:
        zone: mynewzone.sandbox.example.com
        state: present
        vpcs:
          - id: vpc-123456
            region: eu-west-2
          - id: vpc-234567
            region: eu-west-2
```

### CODE REVIEW

File: changelogs/fragments/1299-route53_zone-multi-vpc.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - Added support for multiple VPCs to be applied to route53_zone module
```
Comment (mandar242) [inline changelogs/fragments/1299-route53_zone-multi-vpc.yml:2]:
```suggestion
- route53_zone - added support for associating multiple VPCs to route53 hosted zones (https://github.com/ansible-collections/amazon.aws/pull/1300).
```

File: plugins/modules/route53_zone.py [hunk 2]
```diff
@@ -29,6 +29,23 @@
         description:
             - The VPC Region the zone should be a part of (if this is going to be a private zone).
         type: str
+    vpcs:
+        version_added: 5.2.0
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:33]:
```suggestion
        version_added: 6.0.1
```

File: plugins/modules/route53_zone.py [hunk 3]
```diff
@@ -29,6 +29,23 @@
...
+        description:
+            - The VPCs the zone should be a part of (if this is going to be a private zone).
+        type: list
+        elements: dict
+        suboptions:
+            id:
+                description:
+                    - The ID of the VPC
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:41]:
```suggestion
                    - The ID of the VPC.
```

File: plugins/modules/route53_zone.py [hunk 4]
```diff
@@ -112,15 +139,33 @@
     type: bool
     sample: true
 vpc_id:
-    description: id of vpc attached to private hosted zone
+    description: id of the first vpc attached to private hosted zone (use vpcs for all)
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:142]:
```suggestion
    description: id of the first vpc attached to private hosted zone (use vpcs for associating multiple).
```

File: plugins/modules/route53_zone.py [hunk 5]
```diff
@@ -112,15 +139,33 @@
...
     returned: for private hosted zone
     type: str
     sample: "vpc-1d36c84f"
 vpc_region:
-    description: region of vpc attached to private hosted zone
+    description: region of the first vpc attached to private hosted zone (use vpcs for all)
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:147]:
```suggestion
    description: region of the first vpc attached to private hosted zone (use vpcs for assocaiting multiple).
```

File: plugins/modules/route53_zone.py [hunk 6]
```diff
@@ -112,15 +139,33 @@
...
     returned: for private hosted zone
     type: str
     sample: "eu-west-1"
+vpcs:
+    version_added: 5.2.0
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:152]:
```suggestion
    version_added: 6.1.0
```

File: plugins/modules/route53_zone.py [hunk 7]
```diff
@@ -29,6 +29,23 @@
         description:
             - The VPC Region the zone should be a part of (if this is going to be a private zone).
         type: str
+    vpcs:
+        version_added: 6.1.0
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:33]: @Tompage1994 my apologies, this should be 5.3.0

File: plugins/modules/route53_zone.py [hunk 8]
```diff
@@ -112,15 +139,33 @@
...
     type: str
     sample: "vpc-1d36c84f"
 vpc_region:
-    description: region of vpc attached to private hosted zone
+    description: region of the first vpc attached to private hosted zone (use vpcs for assocaiting multiple).
     returned: for private hosted zone
     type: str
     sample: "eu-west-1"
+vpcs:
+    version_added: 6.1.0
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:152]:
```suggestion
    version_added: 5.3.0
```

File: plugins/modules/route53_zone.py (hunk 7 above)
Comment (tremble) [inline plugins/modules/route53_zone.py:33]:
```suggestion
        version_added: 5.3.0
```

**Comment by gravesm** [comment]
recheck

**Comment by mandar242** [comment]
recheck

File: plugins/modules/route53_zone.py [hunk 9]
```diff
@@ -73,6 +90,16 @@
     vpc_region: us-west-2
     comment: developer domain
 
+- name: create a private zone across 2 VPCs
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:93]:
```suggestion
- name: create a private zone with multiple associated VPCs
```

File: plugins/modules/route53_zone.py [hunk 10]
```diff
@@ -112,15 +139,33 @@
...
+    returned: for private hosted zone
+    type: list
+    elements: dict
+    sample: "[{'id': 'vpc-123456', 'region': 'us-west-2'}]"
+    contains:
+        id:
+            description: ID of the VPC
+            returned: for private hosted zone
+            type: str
+            sample: 'vpc-123456'
```
Comment (mandar242) [inline plugins/modules/route53_zone.py:163]:
```suggestion
            sample: "vpc-123456"
```

**Comment by mandar242** [comment]
recheck

**Comment by mandar242** [comment]
recheck

**Comment by mandar242** [comment]
recheck

**Review by mandar242 — APPROVED** [review]
Thank you @Tompage1994 for contributing on this! Looks good to me!

**Review by alinabuzachis — APPROVED** [review]

**Review by tremble — COMMENTED** [review]
@alinabuzachis I've had a quick look, and nothing's jumping out at me.  If you're good with this one then "mergeit"

**Pull request merged**